src/debug-generator/debug-generator.c

```diff
@@ -142,6 +159,32 @@ static int generate_wants_symlinks(void) {
         return r;
 }
 
+static int install_debug_shell_dropin(const char *tty) {
+        const char *dropin = "/run/systemd/system/debug-shell.service.d/50-tty.conf";
+        const char *dir = "/run/systemd/system/debug-shell.service.d/";
```

Comment:
never write to these dirs form a generator. The generators have their own directories to write to which are life-cycled on "systemctl daemon-reload".